packages/davinci-client/README.md

```diff
@@ -77,67 +159,196 @@ export interface SingleValueCollector {
 
 ```ts
-const updater = davinci.update(collector.id); // Returns an updated for the specific collector
-updater(value); // Will update the store with the value passed in
+// Example SubmitCollector mapping
+const collectors = davinci.collectors();
+collectors.map((collector) => {
+  if (collector.type === 'SubmitCollector') {
+    renderSubmitCollector(
+      collector, // This is the only argument you will need to pass
+    );
+  }
+});
 ```
 
-Normally, you'll use `davinci.collectors()` and it will return all the collectors for that step:
+We will cover the associated action related to this collector in the next section: Continuing a DaVinci Flow below.
+
+#### FlowCollector (Changing a DaVinci flow)
+
+If a user selects an alternative flow button, like Reset Password or Registration. This actikon is associated with a `FlowCollector`, which instructs the application to change from the current flow and start a new, different flow.
+
+To do this, you call the `flow` method on the `davinciClient` passing the `key` property to identify the new flow. This returns a function you can call when the user interacts with it.
 
 ```ts
+// Example FlowCollector mapping
 const collectors = davinci.collectors();
 collectors.map((collector) => {
-  component(collector, davinci.update(collector.id));
+  if (collector.type === 'FlowCollector') {
+    renderFlowCollector(collector, davinciClient.flow(collector));
+  }
 });
 ```
 
-If, at anytime, the dev needs the full state object, they can call the below:
+```ts
+// Example FlowCollector Component
+function renderFlowCollector(collector, startFlow) {
+  // ... component logic
+
+  function onClick(event) {
+    startFlow();
+  }
+
+  // render code
+}
+```
+
+### Continuing a DaVinci flow
+
+After collecting the needed data, you proceed to the next node in the DaVinci flow by calling the `.next()` method on the same `davinci` client object. This can be the result of a user clicking on the button rendered from the `SubmitCollector`, from the "submit" event of the HTML form itself, or from programmatically triggering the submission in the application layer.
 
 ```ts
-const fullState = davinci.getState();
+let nextStep = davinci.next();
 ```
 
-The interface of the full state object looks like this:
+Note: There's no need to pass anything into the `next` method as the DaVinci Client internally stores the updated object needed for the server.
+
+Once the server responds, you will receive the same "node" that will be one of the four types discussed above. You will want to do the same conditional checks to render the appropriate UI.
+
+#### Handling an error
+
+An "error" in the DaVinci Client is caused by data that can to be fixed and resubmitted. A few examples are an email value with an invalid format or a new password that is too short. This is different than a `failure`, which cannot be resubmitted; the flow has to be restarted (this will be covered later in this document).
```

Comment:
Should that change from "...can to be fixed" to "...can be fixed"? (I'm not sure btw...)